src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,58 @@
+package duke;
+import java.util.Scanner;
+
+import duke.command.Command;
+import duke.managers.FileManager;
+import duke.managers.TaskList;
+import duke.misc.Quote;
+
+public class Duke {
+    private TaskList userTaskList;
+    private FileManager fileManager;
+    private Ui ui;
+
+    /**
+     * Duke Constructor
+     *
+     * @param savedTasksPath final path of saved tasks file
+     */
+    public Duke(String savedTasksPath) {
+        this.userTaskList = new TaskList();
+        this.fileManager = new FileManager(savedTasksPath);
+        ui = new Ui();
```

Comment:
Some inconsistencies in the use of "this", understood from my TA that the practice for this mod would be to omit the use of "this" if not needed!